Modules/SubModule_Import/xASL_bids_BIDSifyCheckTimeEncoded.m

```diff
@@ -63,28 +63,22 @@
         if bTimeEncodedFME
             startDetails = regexp(jsonOut.SeriesDescription,'\d\d(_)\d\d(_TIs_)\d\d(_TEs)', 'once');
             jsonOut.TimeEncodedMatrixSize = xASL_str2num(jsonOut.SeriesDescription(startDetails:startDetails+1));
-            jsonOut.TimeEncodedNumberPLD = xASL_str2num(jsonOut.SeriesDescription(startDetails+3:startDetails+4));
+            jsonOut.TimeEncodedNumberTI = xASL_str2num(jsonOut.SeriesDescription(startDetails+3:startDetails+4));
             jsonOut.TimeEncodedNumberTE = xASL_str2num(jsonOut.SeriesDescription(startDetails+10:startDetails+11));
-            fprintf('FME sequence, Hadamard-%d encoded images, %d PLDs, %d TEs\n', jsonOut.HadamardType, jsonOut.HadamardNumberPLD, jsonOut.HadamardNumberTE);
+            fprintf('FME sequence, Hadamard-%d encoded images, %d TIs, %d TEs\n', jsonOut.TimeEncodedMatrixSize, jsonOut.TimeEncodedNumberTI, jsonOut.TimeEncodedNumberTE);
         end
-    else
-        bTimeEncodedFME = false;
     end
     
     % Check total acquired pairs for time encoded sequences
-    if ~isfield(jsonOut,'TotalAcquiredPairs')
-        
-        % Hadamard-4: 1 control / 3 label = 1 pair
-        % Hadamard-8: 1 control / 7 label = 1 pair
-        % etc.
+    if bTimeEncoded
         
-        % Determine number of acquired pairs from image size and number of TE/TI
-        jsonOut.TotalAcquiredPairs = '?';
+        NumberTEs = jsonOut.TimeEncodedNumberTE;
+        NumberPLDs = jsonOut.TimeEncodedNumberTI + 1; % At this stage, jsonOut.PostLabelingDelay has the repeated PLDs already.
         
+        NumberRepetitions = jsonOut.AcquisitionMatrix / (NumberTEs * NumberPLDs);
+        jsonOut.TotalAcquiredPairs = jsonOut.TimeEncodedMatrixSize * NumberRepetitions;
```

Comment:
Nice solution. It probably works perfectly fine, but you could add `if isfield(jsonOut,'TimeEncodedNumberTE') && isfield(jsonOut,'TimeEncodedNumberTE') && TimeEncodedNumberTI` just to be 100% safe, so:

```matlab
if isfield(jsonOut,'TimeEncodedNumberTE') && isfield(jsonOut,'TimeEncodedNumberTE') && TimeEncodedNumberTI
        NumberTEs = jsonOut.TimeEncodedNumberTE;
        NumberPLDs = jsonOut.TimeEncodedNumberTI + 1; % At this stage, jsonOut.PostLabelingDelay has the repeated PLDs already.

        NumberRepetitions = jsonOut.AcquisitionMatrix / (NumberTEs * NumberPLDs);
        jsonOut.TotalAcquiredPairs = jsonOut.TimeEncodedMatrixSize * NumberRepetitions;
end
```